src/utils.js

```diff
@@ -7,25 +9,61 @@ export function extractLaunchParams() {
 }
 
 export function handleLaunch(params) {
+  console.info('handleLaunch', params);
+
   // We use our custom "target" param, since launches with "contentTarget"
   // parameter do not respect "handlesRelaunch" appinfo option. We still
   // fallback to "contentTarget" if our custom param is not specified.
   //
   let { target, contentTarget = target } = params;
+  let href;
+
+  switch (typeof contentTarget) {
+    case 'string': {
+      if (contentTarget.indexOf(YT_BASE_URL.origin) === 0) {
+        console.info('Launching from direct contentTarget');
+        href = contentTarget;
+      } else {
+        // Out of app dial launch with second screen on home: { contentTarget: 'pairingCode=<UUID>&theme=cl&dialLaunch=watch' }
+        console.info('Launching from partial contentTarget');
+        if (contentTarget.indexOf('v=v=') === 0)
+          contentTarget = contentTarget.substring(2);
+
+        href = YT_BASE_URL.toString() + '?' + contentTarget;
+      }
+      break;
+    }
+    case 'object': {
+      console.info(
+        'Voice Launch',
+        contentTarget.intent,
+        contentTarget.intentParam
+      );
 
+      // Ctrl+F tvhtml5LaunchUrlComponentChanged & REQUEST_ORIGIN_GOOGLE_ASSISTANT in base.js for info
+      // TODO: implement google assistant
+      const search = new URLSearchParams();
```

Comment:
We are currently still successfully (from what I could tell) supporting webOS 1.x and 2.x. The issue is that `URLSearchParams` and `URL` WHATWG APIs are not supported there. We could either [try using some shims](https://github.com/jsdom/whatwg-url), or attempt at writing our own SearchParams serializer (which I think may be much easier, since it's just the matter of:
```js
const buildSearchParams = (params) => params.map(([k, v]) => `${encodeURIComponent(k)}=${encodeURIComponent(v)}`).join('&');

// > buildSearchParams([['a', 'b'], ['test', 'ęąćłtest'], ['edge"', '&case']])
// 'a=b&test=%C4%99%C4%85%C4%87%C5%82test&edge%22=%26case'
```

This one shall be properly handled by babel and corejs shims we already use.

Reply:
URL and URLSearchParams should already be automatically pollyfilled by core-js. If not, then it is a build config issue.

https://github.com/zloirock/core-js/blob/master/packages/core-js/modules/web.url-search-params.js
https://github.com/zloirock/core-js/blob/master/packages/core-js/modules/web.url.js

Reply:
Oh indeed, that is indeed correct, I've mistaken that for a different project based on EnyoJS...